### PR TITLE
recent fixes from quietly-turning's Lua-for-SM5 project

### PR DESCRIPTION
**Pull request by quietly-turning**
Hello!

Thanks for working on itgmania; it seems like a nice project.

It looks like `lua-for-itgmania` forked off [my Lua-for-SM5 project](https://github.com/quietly-turning/Lua-For-SM5) sometime in March 2022.  I've continued to clarify and add additional code examples to LuaDocumentation.xml since then, when I have spare time, and recently submitted those changes as a PR to upstream stepmania in https://github.com/stepmania/stepmania/pull/2211

This does the same for your lua-for-itgmania project.   Two of these commits reference Google Analytics, but I removed that code during conflict resolution when merging in 2b9ac9c7e52a985bf733e635d1494bee640704c6.

I can squash into a single commit if you prefer; this was just the least amount of effort on my part to get a pull request submitted. :^)

### CODE REVIEW

File: public/Luadoc/LuaDocumentation.xml [hunk 1]
```diff
@@ -4762,11 +4836,11 @@ prev_note_name, succeeded = options:NoteSkin("cel")
 		Returns the profile for player <code>pn</code>.
 	</Function>
 	<Function name='GetProfileDir' return='string' arguments='ProfileSlot slot'>
-		Returns the profile directory for the provided <Link class="ENUM" function="ProfileSlot">ProfileSlot</Link>, formatted like <code>/Save/LocalProfiles/00000001/</code>.<br />
-		Returns an empty string if the provided <Link class="ENUM" function="ProfileSlot">ProfileSlot</Link> does not have a <Link class="Profile" /> loaded.
+		Returns the profile directory for the provided <Link class='ENUM' function='ProfileSlot'>ProfileSlot</Link>, formatted like <code>/Save/LocalProfiles/00000001/</code>.<br />
+		Returns an empty string if the the provided <Link class='ENUM' function='ProfileSlot'>ProfileSlot</Link> does not have a <Link class='Profile' /> loaded.
```
Comment (teejusb) [inline public/Luadoc/LuaDocumentation.xml]: extra `the`

also `does not have a loaded.` (I think should be `is not loaded` ?)

**Comment by teejusb** [comment]
Personally I think squashing (or condensing to the commits that directly affect the lua docs -- so unlike the `npm update` ones) is preferable but will wait for natano to respond.

File: public/Luadoc/LuaDocumentation.xml (hunk 1 above)
Comment (quietly-turning) [reply public/Luadoc/LuaDocumentation.xml]: Good catch on the extra `the`.  I'll remove that.

---

> also `does not have a loaded.` (I think should be `is not loaded` ?)

That text gets rendered as

> Returns an empty string if the the provided [ProfileSlot](https://itgmania.github.io/lua-For-itgmania/LuaAPI#Enums-ProfileSlot) does not have a [Profile](https://itgmania.github.io/lua-For-itgmania/LuaAPI#Classes-Profile) loaded.

Screen Shot 2022-06-20 at 10 23 11 AM

so it's as-intended.

Comment (teejusb) [reply public/Luadoc/LuaDocumentation.xml]: OH I misread. Thanks for the clarification!

**Comment by natano** [comment]
Merged and squashed down a bit.

**Comment by quietly-turning** [comment]
Thanks @natano!

**Pull request closed without merging**